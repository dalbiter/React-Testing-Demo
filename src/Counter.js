import React, { useState } from 'react';

function Counter() {
    const [count, setCount] = useState(0);
    const increment = () => setCount(count+2)

    return (
        <div>
            <h1>Let's Count!</h1>
            <h2>Current Count: {count}</h2>
            <button onClick={increment}>Add</button>
            <label htmlFor="usr">Username</label>
            <input id="usr" type="text" placeholder="username" />
        </div>
    )
};

export default Counter;