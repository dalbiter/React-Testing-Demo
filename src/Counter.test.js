import { fireEvent, render, screen } from '@testing-library/react';
import Counter from './Counter';

//smoke test
test('it redners without crashing', () => {
    render(<Counter />)
});

//snapshot test
it('should match snapshot', () => {
  const {asFragment} = render(<Counter />)
  expect(asFragment()).toMatchSnapshot()
})

//playing with querying
test('playing with queries', () => {
const { getAllByText, 
        getByText, 
        queryByText,
        getByPlaceholderText,
        getAllByLabelText } = render(<Counter />)
// console.log(getAllByText("Count", {exact: false}))
// console.log(getByText('Cake'))
// console.log(queryByText('Cake'))
// console.log(getByPlaceholderText('username'))
// console.log(getAllByLabelText('Username'))
});

it('button increments counter', () => {
  const { getByText, debug } = render(<Counter />) 
  const h2 = getByText('Current Count: 0')
  const btn = getByText('Add')
  fireEvent.click(btn)
  debug();
  expect(h2).toHaveTextContent('1')// This tests if it goes from 0 to one but it may be good idea to add a test like the one below
  expect(h2).not.toHaveTextContent('0')
})