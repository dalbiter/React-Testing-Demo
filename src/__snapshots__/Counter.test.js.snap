// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Let's Count!
    </h1>
    <h2>
      Current Count: 0
    </h2>
    <button>
      Add
    </button>
    <label
      for="usr"
    >
      Username
    </label>
    <input
      id="usr"
      placeholder="username"
      type="text"
    />
  </div>
</DocumentFragment>
`;
